doc/releases/changelog-dev.md

```diff
@@ -50,6 +50,9 @@
   [(#1017)](https://github.com/PennyLaneAI/catalyst/pull/1017)
   [(#1130)](https://github.com/PennyLaneAI/catalyst/pull/1130)
 
+* Treat `qml.devices.DefaultExecutionConfig` as if it were immutable and make copies of it when
+  modifications are required.
```

Comment:
We should always mention _why_ a change was made :)

Reply:
Fixed! :slightly_smiling_face: